### PR TITLE
Fix MFA module name not showing on step completion

### LINKED ISSUE #18918: TOTP setup message is truncated
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

The TOTP setup success message is truncated.

### Describe the behavior you expected

The dialog box says "Success! Setup done for \<my name>" or something like that

### Steps to reproduce the issue

1. Visit https://\<HA base URL>/profile
2. Scroll down to the MFA modules section
3. Click "Enable" next to TOTP
4. Go through the setup flow
5. Notice the success message is borked

### What version of Home Assistant Core has the issue?

core-2023.11.3

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

Firefox Nightly 122.0a1 built 2023-12-04

### Which operating system are you using to run this browser?

macOS Sonoma 14.1.2

### State of relevant entities

```txt
N/A
```

### Problem-relevant frontend configuration

```yaml
N/A
```

### Javascript errors shown in your browser console/inspector

```txt
Nothing in the JS console, and only 200s in the Network tab of devtools.
```

### Additional information

Screenshot:

Screenshot 2023-12-05 at 17 09 04

Also reproducible on the same OS, Google Chrome 119.0.6045.199 (which, note, is _technically_ not the latest version of Chrome although it [basically is](https://chromereleases.googleblog.com/2023/11/early-stable-update-for-desktop.html)).

### PR DESCRIPTION
**Pull request by codyc1515**
## Proposed change
MFA module name is not showing on step completion resulting in an incomplete and potentially confusing sententence. Videos from three years ago show the message showing "Setup done for Authenticator app". Today, it simply shows "Setup done for".

## Type of change
- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration
1. Enable MFA

Final step does not show the module configured and so the sentence is incomplete.

## Additional information
- This PR fixes or closes issue: fixes #18918
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist
- [ ] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/profile/dialog-ha-mfa-module-setup-flow.ts [hunk 1]
```diff
@@ -97,7 +97,7 @@ class HaMfaModuleSetupFlow extends LitElement {
                   ? html`<p>
                       ${this.hass.localize(
                         "ui.panel.profile.mfa_setup.step_done",
-                        { step: this._step.title }
+                        { step: this._step!.handler }
```
Comment (bramkragten) [inline src/panels/profile/dialog-ha-mfa-module-setup-flow.ts:100]: handler is a technical key, why don't we get title anymore? I don't think this is the right fix.

Comment (codyc1515) [reply src/panels/profile/dialog-ha-mfa-module-setup-flow.ts:100]: I agree with both your comments here. I spent entirely too long longer at this before eventually throwing in the towel and going with this approach, if not just to get a second opinion.

It looks like this was working (from looking at some YouTube videos) in 2018 and was then broken at some point in the years gone by - unnoticed - probably when there was an architecture change.

I thought too that we could probably, for now, safely just refer to a single translation as there is only one method.

Then, I found out that “Authenticator app” is hardcoded somewhere deep in the code (not translated)...

On your second point. This all started with me simply trying to take {qr_code} and wrap it with {url} to get a clickable method of enrolling MFA.

Don’t know if you’ve ever tried scanning a QR on mobile from your own screen (hint: it’s physically impossible, without another device).

With the current approach, I wasn’t able to get there :(

Comment (bramkragten) [reply src/panels/profile/dialog-ha-mfa-module-setup-flow.ts:100]: > Don’t know if you’ve ever tried scanning a QR on mobile from your own screen (hint: it’s physically impossible, without another device).

I always make a screenshot and then run the image through a QR scanner 🙃

Comment (bramkragten) [reply src/panels/profile/dialog-ha-mfa-module-setup-flow.ts:100]: I'll check what is available and what was changed soon

Comment (codyc1515) [reply src/panels/profile/dialog-ha-mfa-module-setup-flow.ts:100]: Just checked and this is working.

Comment (bramkragten) [inline src/panels/profile/dialog-ha-mfa-module-setup-flow.ts:100]:
```suggestion
                        { step:  this._step.title || this._step.handler }
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**